File: src/main/java/com/group/educate/educate/Model/Job/FreeLanceProject.java
//Made By Eyad

package com.group.educate.educate.Model.Job;

import java.lang.instrument.IllegalClassFormatException;

public class FreeLanceProject extends JobPosting{

    private int Duration;
    private int Payout;



    public FreeLanceProject(String JobDescription, String JobName, String JobLocation, String JobRequirements, String JobTitle, PostingType jobPostingType,int Duration,
   int Payout) {
        super(JobDescription, JobName, JobLocation,JobRequirements, JobTitle,jobPostingType);
        this.Duration = Duration;
        this.Payout = Payout;
    }

    public int getDuration() {
        return Duration;
    }

    public int getPayout() {
        return Payout;
    }
    // Access JobPosting id via getter
    public int getPostingId() {
        return super. getJobPostingID();
    }

    public String toString(){
        return "FreeLanceProject{" +
                "Duration=" + Duration +
                ", Payout=" + Payout +
                ", JobPostingID=" + getJobPostingID() +
                '}';
    }
    public static void main(String[] args) throws IllegalClassFormatException {
        FreeLanceProject project = new FreeLanceProject("Build a website", "Web Developer", "Remote", "HTML, CSS, JS", "Website Project", PostingType.FreeLanceProject,1,15);
        FreeLanceProject project2 = new FreeLanceProject("Build a website", "Web Developer", "Remote", "HTML, CSS, JS", "Website Project", PostingType.FreeLanceProject,1,15);
        System.out.println(project.toString());
        System.out.println(project2.toString());
    }

}

File: Frontend/app/Model/Jobs/FreelanceProject.ts
import type { Company } from "../Company";
import type { Recruiter } from "../Users/Recruiter";
import { JobPosting } from "./JobPosting";

export class FreelanceProject extends JobPosting {

    duration: number;
    payout: number;
    jobLocation: String;

    constructor(id: bigint, jobDescription: String, jobRequirements: String, jobName: String, company: Company, jobLocation: String, postingDate: Date, recruiter: Recruiter, duration: number, payout: number) {
        super(id, jobDescription, jobRequirements, jobName, company, jobLocation, postingDate, recruiter,"FreelanceProject");
        this.duration = duration;
        this.payout = payout;
        this.jobLocation = jobLocation;
    }
}